### PR TITLE
Task jd/j334 implement playlists id page

### PR DESCRIPTION
**Pull request by JuliaDzyuba**

### CODE REVIEW

File: backend/src/data/repositories/episode/episode.repository.ts [hunk 1]
```diff
@@ -107,6 +108,26 @@ class Episode {
...
   }
+
+  public getPlaylistEpisodesByQueryByPLaylistId({ playlistId, filter }: LoadPlaylistEpisodesPayload): Promise<TEpisodeWithPodcast[]> {
+    const { limit, offset } = filter;
+
+    return this.#EpisodeModel
+      .query()
+      .withGraphJoined('[podcast]')
+      .joinRelated('[playlistEpisodes]')
+      .where('playlists_episodes.playlist_id', playlistId)
```
Comment (Sayonarko) [inline backend/src/data/repositories/episode/episode.repository.ts]:
```suggestion
      .where('playlistEpisodes.playlist_id', playlistId)
```

File: backend/src/data/repositories/episode/episode.repository.ts [hunk 2]
```diff
@@ -107,6 +108,26 @@ class Episode {
...
+      .limit(limit)
+      .offset(offset);
+  }
+
+  public getPlaylistEpisodesCountByUserId(id: number): Promise<number> {
+    return this.#EpisodeModel
+      .query()
+      .joinRelated('[playlistEpisodes]')
+      .where('playlists_episodes.playlist_id', id)
```
Comment (Sayonarko) [inline backend/src/data/repositories/episode/episode.repository.ts]:
```suggestion
      .where('playlistEpisodes.playlist_id', id)
```

File: backend/src/services/episode/episode.service.ts [hunk 3]
```diff
@@ -353,6 +355,16 @@ class Episode {
       totalCount,
     };
   }
+
+  public async getPlaylistEpisodesByQueryByPLaylistId(payload: LoadPlaylistEpisodesPayload): Promise<PlaylistEpisodeResponse> {
```
Comment (what1s1ove) [inline backend/src/services/episode/episode.service.ts]: we don't need pagination on the page

File: backend/src/data/repositories/episode/episode.repository.ts [hunk 4]
```diff
@@ -110,6 +111,26 @@ class Episode {
...
+    return this.#EpisodeModel
+      .query()
+      .withGraphJoined('[podcast]')
+      .joinRelated('[playlistEpisodes]')
+      .where('playlistEpisodes.playlist_id', playlistId)
+      .limit(limit)
+      .offset(offset);
+  }
+
+  public getPlaylistEpisodesCountByUserId(id: number): Promise<number> {
```
Comment (what1s1ove) [inline backend/src/data/repositories/episode/episode.repository.ts]: we don't pagination for this page

File: backend/src/data/repositories/episode/episode.repository.ts [hunk 5]
```diff
@@ -110,6 +111,26 @@ class Episode {
       .where('favourites.user_id', id)
       .resultSize();
   }
+
+  public getPlaylistEpisodesByQueryByPLaylistId({ playlistId, filter }: LoadPlaylistEpisodesPayload): Promise<TEpisodeWithPodcast[]> {
```
Comment (what1s1ove) [inline backend/src/data/repositories/episode/episode.repository.ts]:
```suggestion
  public getAllByPlaylistId({ playlistId, filter }: LoadPlaylistEpisodesPayload): Promise<TEpisodeWithPodcast[]> {
```

File: frontend/src/store/playlist/reducer.ts [hunk 6]
```diff
@@ -1,21 +1,52 @@
...
   playlists: Playlist[];
   user: User | null;
 };
 
 const initialState: State = {
   dataStatus: DataStatus.IDLE,
+  playlist: null,
+  episodes: [],
+  episodesDataStatus: DataStatus.IDLE,
+  totalCount: 0,
```
Comment (what1s1ove) [inline frontend/src/store/playlist/reducer.ts]:
```suggestion
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/services/episode/episode.service.ts (hunk 3 above)
Comment (what1s1ove) [inline backend/src/services/episode/episode.service.ts]:
```suggestion
  public async getAllByPlaylistId(payload: LoadPlaylistEpisodesPayload): Promise<PlaylistEpisodeResponse> {
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/data/repositories/episode/episode.repository.ts [hunk 7]
```diff
@@ -110,6 +111,26 @@ class Episode {
...
+    const { limit, offset } = filter;
+
+    return this.#EpisodeModel
```
Comment (what1s1ove) [inline backend/src/data/repositories/episode/episode.repository.ts:115]: also, return only public episodes

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: shared/src/common/types/playlist-episode/playlist-episode-response.type.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import { PlaylistEpisodePayloadKey } from '~/common/enums/enums';
+import { EpisodeWithPodcast } from '~/common/types/types';
+
+type PlaylistEpisodeResponse = {
```
Comment (what1s1ove) [inline shared/src/common/types/playlist-episode/playlist-episode-response.type.ts]: delete it

**Review by what1s1ove — CHANGES_REQUESTED** [review]
remove all thing that you don't need

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**